maps/_map.dm

```diff
@@ -176,6 +180,28 @@ var/global/list/accessable_z_levels = list()
 	movementJammed = 1
 	base_turf = /turf/unsimulated/beach/sand
 
+/datum/zLevel/snow
+	name = "snow"
+	base_turf = /turf/unsimulated/floor/snow
+	movementChance = ZLEVEL_BASE_CHANCE * ZLEVEL_SPACE_MODIFIER
+
+/datum/zLevel/snow/post_mapload()
+	var/lake_density = rand(2,8)
+	for(var/i = 0 to lake_density)
+		var/turf/T = locate(rand(0, world.maxx),rand(0, world.maxy), z)
+		if(!istype(T, base_turf))
+			continue
+		var/generator = pick(typesof(/obj/structure/radial_gen/cellular_automata/ice))
+		new generator(T)
+
+	var/tree_density = rand(25,45)
+	for(var/i = 0 to tree_density)
+		var/turf/T = locate(rand(0,world.maxx),rand(0, world.maxy), z)
+		if(!istype(T, base_turf))
```

Comment:
If we're unlucky and they pick a base turf 25 to 45 times, we'll simply never have any forest nor lakes
Maybe there is a smarter way to do this (define `outside of the station` as an area and pick a turf within this area)

Reply:
Hmm, tried that before with checking all turfs in /area/space, and it got *odd*

Reply:
Odd how

Reply:
Odd as in it would either lock up, or not work.

Had an idea, could exclude the center of the map, so rather than 1,500 it's like pick(rand(1,200),rand(300,500)) so there's always a deadzone.

Not specifically those numbers, but you get my drift, right?